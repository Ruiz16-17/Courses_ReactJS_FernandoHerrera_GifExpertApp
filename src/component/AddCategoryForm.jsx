import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const AddCategoryForm = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const handleSubmit = (event) => {

        event.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue,...categories]);
            setInputValue('');
        }
        
    }

    return (

        <form onSubmit={handleSubmit}>

            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>

    )
}

AddCategoryForm.propType = {
    setCategories : PropTypes.func.isRequired
}