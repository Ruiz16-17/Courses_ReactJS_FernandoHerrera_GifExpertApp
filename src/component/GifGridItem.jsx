import PropTypes from 'prop-types';
import React from 'react';

export const GifGridItem = ({id, url, title}) => {
    return (
        <div 
            key={id}
            className="card animate__animated animate__bounce"
        >
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propType = {
    id : PropTypes.string.isRequired,
    title : PropTypes.string.isRequired,
    url :PropTypes.string.isRequired
    
}