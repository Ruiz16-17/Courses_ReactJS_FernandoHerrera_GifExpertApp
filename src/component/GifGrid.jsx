import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {loading, data: images} = useFetchGif(category);

    return (
        <>

            <h3 className="animate__animated animate__flash">{category}</h3>

            {loading && <p>Loading...</p> }

            <div className="card-grid">
            
                {images.map(image => {

                    return (
                        <GifGridItem
                            key={image.id}
                            {...image}

                        />
                    )
                })}

            </div>
        </>
    )
}

GifGrid.propType = {
    category : PropTypes.string.isRequired
    
}