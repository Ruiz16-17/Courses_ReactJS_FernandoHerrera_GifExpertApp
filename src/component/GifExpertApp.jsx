import React, { useState } from 'react';
import { AddCategoryForm } from './AddCategoryForm';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            
            <AddCategoryForm setCategories={setCategories}/>
            
            <hr/>

            <ol>
                {
                    categories.map(
                        category => {
                            return <GifGrid 
                                key={category}
                                category={category} />
                        }
                    )
                
                }
            </ol>

        </>
    )
}
