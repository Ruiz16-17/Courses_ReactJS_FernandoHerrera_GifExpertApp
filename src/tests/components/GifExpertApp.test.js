import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import React from "react";
import { GifExpertApp } from '../../component/GifExpertApp';
describe('Test on <GifExpertApp/>', () => {
    
    
    test('Should show correctly', () => {
        const wrapper = shallow(<GifExpertApp />); 
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show a categoryies list', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>); 
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
})
