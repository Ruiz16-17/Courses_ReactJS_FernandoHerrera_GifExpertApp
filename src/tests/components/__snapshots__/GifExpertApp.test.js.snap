// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp/> Should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryForm
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on <GifExpertApp/> should show a categoryies list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryForm
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
